service/app/controllers/BaseController.java

```diff
@@ -126,4 +134,133 @@ public Request createSBRequest(play.mvc.Http.Request httpReq, String operation)
     request.setOperation(operation);
     return request;
   }
+  
+  /**
+   * This method will create failure response
+   *
+   * @param request Request
+   * @param code ResponseCode
+   * @param headerCode ResponseCode
+   * @return Response
+   */
+  public static Response createFailureResponse(
```

Comment:
@Hari-stackroute @reshmi-nair  - why are these methods added now in this PR? are there no failure responses sent for groups APIs before this auth checks? If so, how is the failure response created in such cases.

Reply:
@vrayulu : as of how due to time constraint we have taken the dependency class and methods from learner to here.
Here we are creating the authentication failure message as response.

Reply:
@Hari-stackroute - where are we creating the other failure messages in groups API? because of time constraints, we will go into this trap of duplicate code & accumulate a lot of tech debt. We are starting group service from scratch now and we should implement it the correct way.